modules/gapi/samples/face_detection.cpp

```diff
@@ -0,0 +1,758 @@
+#include <algorithm>
+#include <cctype>
+#include <cmath>
+#include <iostream>
+#include <limits>
+#include <numeric>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+#include <opencv2/gapi.hpp>
+#include <opencv2/gapi/core.hpp>
+#include <opencv2/gapi/imgproc.hpp>
+#include <opencv2/gapi/cpu/gcpukernel.hpp>
+#include <opencv2/gapi/infer.hpp>
+#include <opencv2/gapi/infer/ie.hpp>
+#include <opencv2/gapi/streaming/cap.hpp>
+#include <opencv2/gapi/gopaque.hpp>
+#include <opencv2/highgui.hpp>
+
+const std::string about =
+"This is an OpenCV-based version of OMZ MTCNN Face Detection example";
+const std::string keys =
+"{ h help     |                           | Print this help message }"
+"{ input      |                           | Path to the input video file }"
+"{ mtcnnpm    | mtcnn-p.xml               | Path to OpenVINO MTCNN P (Proposal) detection model (.xml)}"
+"{ mtcnnpd    | CPU                       | Target device for the MTCNN P (e.g. CPU, GPU, VPU, ...) }"
+"{ mtcnnrm    | mtcnn-r.xml               | Path to OpenVINO MTCNN R (Refinement) detection model (.xml)}"
+"{ mtcnnrd    | CPU                       | Target device for the MTCNN R (e.g. CPU, GPU, VPU, ...) }"
+"{ mtcnnom    | mtcnn-o.xml               | Path to OpenVINO MTCNN O (Output) detection model (.xml)}"
+"{ mtcnnod    | CPU                       | Target device for the MTCNN O (e.g. CPU, GPU, VPU, ...) }"
+"{ thrp       | 0.6                       | MTCNN P confidence threshold}"
+"{ thrr       | 0.7                       | MTCNN R confidence threshold}"
+"{ thro       | 0.7                       | MTCNN O confidence threshold}"
+"{ half_scale | false                     | MTCNN P use half scale pyramid}"
+;
+
+namespace {
+std::string weights_path(const std::string& model_path) {
+    const auto EXT_LEN = 4u;
+    const auto sz = model_path.size();
+    CV_Assert(sz > EXT_LEN);
+
+    const auto ext = model_path.substr(sz - EXT_LEN);
+    CV_Assert(cv::toLowerCase(ext) == ".xml");
+    return model_path.substr(0u, sz - EXT_LEN) + ".bin";
+}
+//////////////////////////////////////////////////////////////////////
+} // anonymous namespace
+
+namespace custom {
+namespace {
+
+// Define custom structures and operations
+#define NUM_REGRESSIONS 4
+#define NUM_PTS 5
+
+struct BBox {
+    double x1;
+    double y1;
+    double x2;
+    double y2;
+
+    cv::Rect getRect() const { return cv::Rect(static_cast<int>(x1), static_cast<int>(y1), static_cast<int>(x2 - x1), static_cast<int>(y2 - y1)); }
+
+    BBox getSquare() const {
+        BBox bbox;
+        double bboxWidth = x2 - x1;
+        double bboxHeight = y2 - y1;
+        double side = std::max(bboxWidth, bboxHeight);
+        bbox.x1 = static_cast<double>(x1) + (bboxWidth - side) * 0.5;
+        bbox.y1 = static_cast<double>(y1) + (bboxHeight - side) * 0.5;
+        bbox.x2 = bbox.x1 + side;
+        bbox.y2 = bbox.y1 + side;
+        return bbox;
+    }
+};
```

Comment:
I don't quite get what is the need in `BBox` as there is [cv::Rect2f](https://docs.opencv.org/4.5.2/dc/d84/group__core__basic.html#ga2d6c1acbbb651cf626f138748aac3d02)

Maybe reusing a standard class would save our efforts (maybe not).